plugins/woocommerce/includes/class-wc-install.php

```diff
@@ -800,12 +786,12 @@ public static function create_pages() {
 				'cart'           => array(
 					'name'    => _x( 'cart', 'Page slug', 'woocommerce' ),
 					'title'   => _x( 'Cart', 'Page title', 'woocommerce' ),
-					'content' => '<!-- wp:shortcode -->[' . $cart_shortcode . ']<!-- /wp:shortcode -->',
+					'content' => self::get_cart_block_content(),
 				),
 				'checkout'       => array(
 					'name'    => _x( 'checkout', 'Page slug', 'woocommerce' ),
 					'title'   => _x( 'Checkout', 'Page title', 'woocommerce' ),
-					'content' => '<!-- wp:shortcode -->[' . $checkout_shortcode . ']<!-- /wp:shortcode -->',
+					'content' => self::get_checkout_block_content(),
```

Comment:
We're effectively deprecated a couple of filters for good, do we need to communicate this somehow? or provider a workaround?

Reply:
While we are replacing those filters, we're not swapping to another shortcode. We're swapping the default to blocks. There are already existing filters to change the entire content (we were consumers of this) so I don't feel a workaround is needed. Just a note.

Reply:
Ok, so this is the approach I suggest to people overriding cart and checkout default content, before they could narrow it down to shortcode tags, or use the `woocommerce_create_pages` filter. I propose graceful deprecate the shortcode tags filters, and point existing extensions/themes in the right direction through the usage of `woocommerce_create_pages`. Then we can address this in a separate PR with proper scoping of removing these.

There is usage scattered around, although only for doc purposes, and also tests that would need to be removed. Furthermore this way we ensure retrocompatibility, and allow a graceful timeframe for 3rd party to adapt.

Something along the lines of this:

```
/**
		 * Determines the cart shortcode tag used for the cart page.
		 *
		 * @since 2.1.0
		 * @deprecated 8.3.0 This filter is deprecated and will be removed in future versions.
		 */
		$cart_shortcode = apply_filters_deprecated( 'woocommerce_cart_shortcode_tag', array( false ), '8.3.0', 'woocommerce_create_pages' );

		$cart_page_content = ! $cart_shortcode ? self::get_cart_block_content() : '<!-- wp:shortcode -->[' . $cart_shortcode . ']<!-- /wp:shortcode -->';

		/**
		 * Determines the checkout shortcode tag used on the checkout page.
		 *
		 * @since 2.1.0
		 * @deprecated 8.3.0 This filter is deprecated and will be removed in future versions.
		 */
		$checkout_shortcode = apply_filters_deprecated( 'woocommerce_checkout_shortcode_tag', array( false ), '8.3.0','woocommerce_create_pages' );

		$checkout_page_content = ! $checkout_shortcode ? self::get_checkout_block_content() : '<!-- wp:shortcode -->[' . $checkout_shortcode . ']<!-- /wp:shortcode -->';
```

Reply:
https://github.com/woocommerce/woocommerce/pull/40867/commits/5a247444d4da2c8e10165738c8092fd7536b598b